Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2018-06-03
Initial Package Version: 0.92.3
Upstream Status: Assumed applied (the changes for 0.65 have certainly
been applied).
Origin: Found at Arch (for poppler-0.64.0), changes for 0.65.0 from
Jan Palus of pld-linux.
Description: Fixes build with poppler-0.65.0.

diff -Naur a/src/extension/internal/pdfinput/pdf-parser.cpp b/src/extension/internal/pdfinput/pdf-parser.cpp
--- a/src/extension/internal/pdfinput/pdf-parser.cpp	2018-03-11 20:38:09.000000000 +0000
+++ b/src/extension/internal/pdfinput/pdf-parser.cpp	2018-06-03 04:58:21.175386795 +0100
@@ -37,8 +37,7 @@
 #include "util/units.h"
 
 #include "goo/gmem.h"
-#include "goo/GooTimer.h"
-#include "goo/GooHash.h"
+#include "goo/GooString.h"
 #include "GlobalParams.h"
 #include "CharTypes.h"
 #include "Object.h"
@@ -2490,7 +2489,7 @@
     builder->updateFont(state);
     fontChanged = gFalse;
   }
-  doShowText(args[0].getString());
+  doShowText((GooString *)args[0].getString());
 }
 
 // TODO not good that numArgs is ignored but args[] is used:
@@ -2511,7 +2510,7 @@
   ty = state->getLineY() - state->getLeading();
   state->textMoveTo(tx, ty);
   builder->updateTextPosition(tx, ty);
-  doShowText(args[0].getString());
+  doShowText((GooString *)args[0].getString());
 }
 
 // TODO not good that numArgs is ignored but args[] is used:
@@ -2534,7 +2533,7 @@
   ty = state->getLineY() - state->getLeading();
   state->textMoveTo(tx, ty);
   builder->updateTextPosition(tx, ty);
-  doShowText(args[2].getString());
+  doShowText((GooString *)args[2].getString());
 }
 
 // TODO not good that numArgs is ignored but args[] is used:
@@ -2572,7 +2571,7 @@
       }
       builder->updateTextShift(state, obj.getNum());
     } else if (obj.isString()) {
-      doShowText(obj.getString());
+      doShowText((GooString *)obj.getString());
     } else {
       error(errSyntaxError, getPos(), "Element of show/space array must be number or string");
     }
@@ -2732,7 +2731,7 @@
 {
   Object obj1, obj2, obj3, refObj;
 
-  char *name = args[0].getName();
+  char *name = (char *)args[0].getName();
 #if defined(POPPLER_NEW_OBJECT_API)
   if ((obj1 = res->lookupXObject(name)).isNull()) {
 #else
diff -Naur a/src/extension/internal/pdfinput/svg-builder.cpp b/src/extension/internal/pdfinput/svg-builder.cpp
--- a/src/extension/internal/pdfinput/svg-builder.cpp	2018-03-11 20:38:09.000000000 +0000
+++ b/src/extension/internal/pdfinput/svg-builder.cpp	2018-06-03 04:58:15.158421367 +0100
@@ -1020,7 +1020,7 @@
     GfxFont *font = state->getFont();
     // Store original name
     if (font->getName()) {
-        _font_specification = font->getName()->getCString();
+        _font_specification = (char *)font->getName()->getCString();
     } else {
         _font_specification = (char*) "Arial";
     }
